### ISSUE TITLE
Way to `teach' NSGA-II with previous data?

### ISSUE THREAD

**Issue by jbae11** [aarongarrett/inspyred#18]
* inspyred version: 1.0
* Python version: 3.7.3
* Operating System: MAC OS

### Description

I am using Inspyred to run NSGA-II to do multiobjective optimization on a very computationally heavy problem. While running, my cluster crashed and I lost that run. I have 40 populations (samples and results), but it seems like I have to start all over again. Is there a way to 'feed' previously generated data into NSGA-II so that it starts off with that knowledge?

Thanks for the great tool!

**Comment by aarongarrett** [comment]
Yes. You can pass a list of candidates as seeds into the evolve method. For
instance, suppose you had two candidates that you wanted to start from: [0,
8, 6, 7] and [5, 3, 0, 9].
Then you could make a call like this to evolve:

myec.evolve(generator=mygen, evaluator=myeval, pop_size=100, seeds=[[0, 8,
6, 7], [5, 3, 0, 9]])

Obviously, if you have lots of stuff, you'd prefer to read them from a file
and build that list programmatically. Also, make sure your pop_size is at
least as large as your number of seed candidates.


--
Aaron Garrett

**Comment by jbae11** [comment]
thanks for the very quick reply. Is it possible to maybe give them the 'solutions' as well?
For example, if I am optimizing a multiobjective function with 4 inputs and 2 outputs, can I input 10 sets of 4 inputs and 2 outputs so that the NSGA-II starts with those datasets 'in mind'?

**Comment by aarongarrett** [comment]
So you also want to give it the fitness values? That's not as simple to do.
We'd have to write some custom code for that. Or maybe you could load a
memoized wrapper with the fitness values for your seed candidates so that
the evaluator wouldn't really have to run on those...

--
Aaron Garrett

**Comment by jbae11** [comment]
I see, so the best way would be to give them my best candidates for the first generation?

**Comment by aarongarrett** [comment]
Well, we could figure out a way to get the fitness values loaded, but it
would take some coding. I guess it depends on how long your evaluations
take. If they only take 30 minutes total to re-evaluate, then that would be
far faster than working out a programmatic solution to your problem.

--
Aaron Garrett

**Comment by jbae11** [comment]
My evaluations take a couple of hours per single evaluation. But I wouldn't want to trouble you with coming up with a new developments.. I bet my use case is quite rare..?

**Comment by aarongarrett** [comment]
Are you able to send me your code to look at? If so, send it to
aaron.lee.garrett@gmail.com


--
Aaron Garrett

**Comment by jbae11** [comment]
I'm afraid not, it's not quite for public release. I'm sorry

**Comment by jbae11** [comment]
Also, can you tell me what's the best way to cite your software? thanks

**Comment by aarongarrett** [comment]
Sure. This is what others have done:
https://groups.google.com/d/msg/inspyred/K43UQsCLitM/7ltTXyDbAQAJ

--
Aaron Garrett
